src/instruction/jk_inst.rs

```diff
@@ -81,27 +81,30 @@ impl Instruction for JkInst {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::parser::Construct;
+    use crate::parser::{Construct, Token};
 
     #[test]
     fn t_invalid_jkinst() {
-        let (_, fc) = Construct::function_call("tamer()").unwrap();
+        let (input, id) = Token::identifier("tamer()").unwrap();
+        let (_, fc) = Construct::function_call(input, &id).unwrap();
```

Comment:
The function identifier should be ``tamer``not ``tamer()``,

Reply:
Token::identifier will parse `tamer` leaving the `()` in input which is then fed to function_call.
This is equivalent to giving `tamer()` directly to the previous implementation of function_call.

The reason why identifier has been extracted from function_call is to reduce backtracking.

Reply:
Token::identifier will parse `tamer` leaving the `()` in input which is then fed to function_call.
This is equivalent to giving `tamer()` directly to the previous implementation of function_call.

The reason why identifier has been extracted from function_call is to reduce backtracking.